### PR TITLE
Various vagrant improvements

### PR DESCRIPTION
**Pull request by jcsalem**
Three main changes:
 1) More consistent logging of vagrant commands
 2) Automatically determine external IP address of Vagrant instance
 3) Retain VAGRANT_CWD setting

### CODE REVIEW

File: lib/rubber/cloud/vagrant.rb [hunk 1]
```diff
@@ -49,23 +47,98 @@ def describe_instances(instance_id=nil)
...
+       run_vagrant_command('resume', instance.instance_id)
       end
+
+      private
+
+      def run_vagrant_command(subcmd, instance_or_id = nil, args = "", return_result = false)
+        cmd = "vagrant #{subcmd}"
+
+        # Setup instance id and VAGRANT_CWD
+        if instance_or_id
```
Comment (james-lawrence) [inline lib/rubber/cloud/vagrant.rb]: this function is overly complicated it should be broken into multiple functions.

File: lib/rubber/cloud/vagrant.rb [hunk 2]
```diff
@@ -49,23 +47,98 @@ def describe_instances(instance_id=nil)
...
+          if instance_or_id.kind_of?(String)
+            instance_id = instance_or_id
+            instance = Rubber.instances[instance_or_id]
+          else
+            instance = instance_or_id
+            instance_id = instance.instance_id
+          end
+
+          if instance && instance.provider_options && instance.provider_options[:vagrant_cwd]
+            cmd = "VAGRANT_CWD=#{instance.provider_options[:vagrant_cwd]} " + cmd
```
Comment (james-lawrence) [inline lib/rubber/cloud/vagrant.rb]: what is cwd?

**Comment by jcsalem** [comment]
cwd is the working directory with the Vagrantfile.
Documented here: 
 http://docs.vagrantup.com/v2/other/environmental-variables.html

It's use is required when creating a Vagrant staging instance:
https://confluence.backupify.org/display/ENG/Vagrant%3A+Provision+a+Staging+Instance+using+a+KVM+VM+Host

I just wanted to change Rubber to retain that setting after the initial deployment. It caused Duy some confusion when it wasn't retained and he left it off when doing a deploy.

**Comment by james-lawrence** [comment]
LGTM

**Pull request merged**